core/src/main/java/org/apache/iceberg/PartitionStats.java

```diff
@@ -78,7 +78,16 @@ public int equalityDeleteFileCount() {
     return equalityDeleteFileCount;
   }
 
+  /**
+   * @deprecated since 1.9.0, will be removed in 1.10.0, use {@link #totalRecordCountOptional()}
+   *     instead.
+   */
+  @Deprecated
```

Comment:
do we really need to deprecate the existing method and introduce a new one to handle this properly?

Reply:
Maybe no one using it as the writer code is not present in previous version. 

But just following the [guidelines](https://iceberg.apache.org/contribute/?h=contri#minor-version-deprecations-required) as it is a public interface and introduced in previous version.

Reply:
> method's return type stays the same

We wanted to return null when not computed as per the discussion. We can't do that if the return type is primitive.